### DISCUSSION TITLE
How to get User Id from Android Iterable SDK after initialization?

### DISCUSSION THREAD

**Discussion by williamspete321** [Iterable/iterable-android-sdk#526] [Q&A] [answered]
We would like to access the userId in a similar way that the iOS SDK has access to user id after initialization. This will let us be in-sync with our iOS counterpart:

**Swift:**

`IterableAPI.userId`

In the Android SDK, I see a userId property but it is protected, and trying to access it the compiler warns us:

**Android/Java:**

`IterableApi.getInstance().getUserId()`

'Cannot access 'getUserId': it is package-private in 'IterableApi'

How can the userId associated with a device be accessed from the Iterable SDK for Android?

**Comment by jena-chakour** [comment]
Thanks for surfacing this, we will take a look at it

**Comment by Ayyanchira** [comment, marked as answer]
Our latest release [3.4.11](https://github.com/Iterable/iterable-android-sdk/releases/tag/3.4.11) has made this method public.
Thanks for the ask :)

**Reply by williamspete321** [reply]
Thank you!
